lib/new_post_manager.rb

```diff
@@ -73,6 +73,7 @@ def self.user_needs_approval?(manager)
 
     (user.trust_level <= TrustLevel.levels[:basic] && user.post_count < SiteSetting.approve_post_count) ||
     (user.trust_level < SiteSetting.approve_unless_trust_level.to_i) ||
+    (manager.args[:title].present? && user.trust_level < SiteSetting.approve_new_topics_unless_trust_level.to_i) ||
```

Comment:
Ruby tests. You'll need to add a test case in [spec/components/new_post_manager_spec.rb](https://github.com/discourse/discourse/blob/master/spec/components/new_post_manager_spec.rb) ;)